tempesta_fw/t/functional/tests/test_frang.py

```diff
@@ -0,0 +1,516 @@
+		self.cfg.add_option('http_field_len', '10')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST /a.html HTTP/1.0\r\nHost: loc\r\n" +\
+b"Content-Type: application/xml\r\n" +\
+b"Content-Length: 20\r\n\r\n" +\
+b"<html>content</html>\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def ct_vals(self):
+		global tcount
+
+		print("ct_vals")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_ct_vals', '[\"text/html\"]')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST / HTTP/1.0\r\nhost: loc\r\n" +\
+b"Content-type: application/xml\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def req_method(self):
+		global tcount
+
+		print("req_method")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_methods', 'get')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST / HTTP/1.0\r\nhost: loc\r\n" +\
+b"Content-type: application/xml\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+	def header_chunks(self):
+		global tcount
+
+		print("header_chunks\n")
+
+		part1 = b'GET / HTTP/1.0\r\n'
+		part2 = b'host: loc\r\n'
+		part3 = b'Connection: close\r\n\r\n'
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_header_chunk_cnt', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		self.s.send(part2)
+		self.s.send(part3)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+		print("res:", self.res)
+
+	def header_timeout(self):
+		global tcount
+		part1 = b'GET / HTTP/1.0\r\n'
+		part2 = b'host: loc\r\n\r\n'
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('client_header_timeout', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		time.sleep(1)
```

Comment:
Probably it's better to sleep bit more than 1 second to exceed the timeout with certainty.